lib/libc/include/sys/types.h

```diff
@@ -44,7 +44,11 @@ enum handler_return {
     INT_RESCHEDULE,
 };
 
+#ifdef __LP64__
 typedef signed long int ssize_t;
+#else
+typedef signed int ssize_t;
```

Comment:
Unfortunately this isn't really always true, it varies by architecture. For LK I've bee forcing size_t to be a long always, even if it's against the default the compiler sets it to, and it works reasonably well as long as the code is mostly using stuff built in tree.

Is there a strong need for this? If so we may be able to set it for particular architectures that its an issue with (such as linux-x86, where i know size_t is unsigned int).